pkg/operator/ceph/pool/radosnamespace/controller.go

```diff
@@ -427,25 +427,25 @@ func (r *ReconcileCephBlockPoolRadosNamespace) deleteRadosNamespace(radosNamespa
 	containsImages, deleteErr := cephclient.DeleteRadosNamespace(r.context, r.clusterInfo, radosNamespace.Spec.BlockPoolName, name)
 	// If deleteErr is not nil, it means the deletion failed, but we still want to
 	// report a condition whether the rados namespace contains images
+	var emptyCondition cephv1.Condition
 	if containsImages {
-		var emptyCondition cephv1.Condition
-		if containsImages {
-			emptyCondition = dependents.DeletionBlockedDueToNonEmptyRadosNSCondition(
-				true,
-				fmt.Sprintf("rados namespace %q contains images or snapshots and cannot be deleted", radosNamespace.Name))
-		} else {
-			emptyCondition = dependents.DeletionBlockedDueToNonEmptyRadosNSCondition(
-				false,
-				fmt.Sprintf("rados namespace %q is empty and can be deleted", radosNamespace.Name))
-		}
-		logger.Info(emptyCondition.Message)
-
-		err := reporting.UpdateStatusConditionsWithRetry(
-			r.opManagerContext, r.client, radosNamespace, nsName, radosNamespace.Kind, emptyCondition)
-		if err != nil {
-			logger.Warningf("failed to update %q status with deletion blocked conditions: %v", nsName.String(), err)
-		}
+		emptyCondition = dependents.DeletionBlockedDueToNonEmptyRadosNSCondition(
+			true,
+			fmt.Sprintf("rados namespace %q contains images or snapshots and cannot be deleted", radosNamespace.Name))
+	} else {
+		emptyCondition = dependents.DeletionBlockedDueToNonEmptyRadosNSCondition(
+			false,
+			fmt.Sprintf("rados namespace %q is empty and can be deleted", radosNamespace.Name))
+	}
+	logger.Info(emptyCondition.Message)
+
+	err := reporting.UpdateStatusConditionsWithRetry(
+		r.opManagerContext, r.client, radosNamespace, nsName, radosNamespace.Kind, emptyCondition)
+	if err != nil {
+		logger.Warningf("failed to update %q status with deletion blocked conditions: %v", nsName.String(), err)
+	}
 
+	if containsImages {
```

Comment:
This is really the only change, moving this condition here, down from line 431. The rest of the changes are just changing of indentation.

Reply:
The `DeletionBlockedDueToNonEmptyRadosNSCondition` condition was not being raised. It's not critical, since in any case the CR will be immediately deleted after that.

Reply:
> This looks a little related to this #16008

As discussed, let's go ahead and merge this, then proceed with #16008